test/testpath.cpp

```diff
@@ -38,6 +38,20 @@ class TestPath : public TestFixture {
         TEST_CASE(is_c);
         TEST_CASE(is_cpp);
         TEST_CASE(get_path_from_filename);
+        TEST_CASE(isMatchingFilter);
+    }
+
+    void isMatchingFilter() const
+    {
+        ASSERT(Path::isMatchingFilter("c:\\foo\\bar\\os_linux.c", "bar"));
+        ASSERT(Path::isMatchingFilter("c:\\foo\\bar\\os_linux.c", "os_win")==false);
+        ASSERT(Path::isMatchingFilter("c:\\foo\\bar\\os_linux.c", "bar"));
+        ASSERT(Path::isMatchingFilter("c:\\foo\\bar\\os_linux.c", "b*"));
+        ASSERT(Path::isMatchingFilter("c:\\foo\\bar\\os_linux.cpp", "*.cpp"));
```

Comment:
In my opinion this should be a match because that is "de facto" behavior for globs:

    ASSERT(Path::isMatchingFilter("foo/bar/file1.cpp", "foo/bar/*.cpp"));